module/mfrc522_module.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include <linux/module.h>
+#include <linux/init.h>
+#include <linux/spi/spi.h>
+#include <linux/regmap.h>
+#include <linux/delay.h>
+
+#include "mfrc522_spi.h"
+
+MODULE_LICENSE("GPL v2");
+MODULE_AUTHOR("ks0n");
+MODULE_DESCRIPTION("Driver for the MFRC522 RFID Chip");
+
+#define MFRC522_VERSION_BASE 0x90
+#define MFRC522_VERSION_1 0x91
+#define MFRC522_VERSION_2 0x92
+#define MFRC522_VERSION_NUM(ver) ((ver) - MFRC522_VERSION_BASE)
+
+
+static int mfrc522_spi_probe(struct spi_device *spi);
+
+static struct spi_driver mfrc522_spi = {
+    .driver = {
+        .name = "mfrc522",
+        .owner = THIS_MODULE,
+    },
+    .probe = mfrc522_spi_probe,
+};
+
+static int mfrc522_detect(struct spi_device *client) {
+    struct address_byte version_reg_read = address_byte_build(MFRC522_SPI_READ, MFRC522_VERSION_REG);
+    char version;
+
+    spi_write(client, &version_reg_read, 1);
+    mdelay(100);
+    spi_read(client, &version, 1);
+
+    switch (version) {
+            case MFRC522_VERSION_1 || MFRC522_VERSION_2:
+		    version = MFRC522_VERSION_NUM(version);
+        	    pr_info("[MFRC522] MFRC522 version %d detected\n", version);
+		    return version;
+            default:
+        	    pr_info("[MFRC522] this chip is not an MFRC522: 0x%x\n", version);
+    }
+    
+    return -1;
+}
+
+
+static int mfrc522_spi_probe(struct spi_device *client)
+{
+    pr_info("[MFRC522] SPI Probed\n");
+
+    if (client->max_speed_hz > MFRC522_SPI_MAX_CLOCK_SPEED) {
+	    pr_info("[MFRC522] Current speed is to high %u. Setting speed to %u\n", client->max_speed_hz, MFRC522_SPI_MAX_CLOCK_SPEED);
+	    client->max_speed_hz = MFRC522_SPI_MAX_CLOCK_SPEED;
+    }
+
+    return mfrc522_detect(client);
+}
+
```

Comment:
We should probably move this into `mfrc522_spi.c`

Reply:
Not lines 15 to 18 and ``mfrc522_detect``, but the rest I agree

Reply:
@n1tram1 and I think we should just wait for a regmap pull request